agent/api/ecsclient/client_test.go

```diff
@@ -269,6 +303,68 @@ func TestRegisterContainerInstance(t *testing.T) {
 	}
 }
 
+func TestRegisterContainerInstance(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockEC2Metadata := mock_ec2.NewMockEC2MetadataClient(mockCtrl)
+	additionalAttributes := map[string]string{"my_custom_attribute": "Custom_Value1",
+		"my_other_custom_attribute": "Custom_Value2",
+	}
+	client, mc, _ := NewMockClient(mockCtrl, mockEC2Metadata, additionalAttributes)
+
+	capabilities := []string{"capability1", "capability2"}
+	expectedAttributes := map[string]string{
+		"ecs.os-type":               api.OSType,
+		"my_custom_attribute":       "Custom_Value1",
+		"my_other_custom_attribute": "Custom_Value2",
+	}
+
+	mockEC2Metadata.EXPECT().ReadResource(ec2.INSTANCE_IDENTITY_DOCUMENT_RESOURCE).Return([]byte("instanceIdentityDocument"), nil)
+	mockEC2Metadata.EXPECT().ReadResource(ec2.INSTANCE_IDENTITY_DOCUMENT_SIGNATURE_RESOURCE).Return([]byte("signature"), nil)
+	mc.EXPECT().RegisterContainerInstance(gomock.Any()).Do(func(req *ecs.RegisterContainerInstanceInput) {
+		assert.Nil(t, req.ContainerInstanceArn)
+		assert.Equal(t, configuredCluster, *req.Cluster, "Wrong cluster")
+		assert.Equal(t, "instanceIdentityDocument", *req.InstanceIdentityDocument, "Wrong IID")
+		assert.Equal(t, "signature", *req.InstanceIdentityDocumentSignature, "Wrong IID sig")
+		assert.Equal(t, 4, len(req.TotalResources), "Wrong length of TotalResources")
+		resource, ok := findResource(req.TotalResources, "PORTS_UDP")
+		assert.True(t, ok, `Could not find resource "PORTS_UDP"`)
+		assert.Equal(t, "STRINGSET", *resource.Type, `Wrong type for resource "PORTS_UDP"`)
+		assert.Equal(t, 5, len(req.Attributes), "Wrong number of Attributes")
+		for i := range req.Attributes {
+			if strings.Contains(*req.Attributes[i].Name, "capability") {
+				assert.Contains(t, capabilities, *req.Attributes[i].Name)
+			} else {
+				assert.Equal(t, expectedAttributes[*req.Attributes[i].Name], *req.Attributes[i].Value)
+			}
+		}
+	}).Return(&ecs.RegisterContainerInstanceOutput{ContainerInstance: &ecs.ContainerInstance{ContainerInstanceArn: aws.String("registerArn"), Attributes: buildAttributeList(capabilities, expectedAttributes)}}, nil)
+
+	arn, err := client.RegisterContainerInstance("", capabilities)
+	if err != nil {
+		t.Errorf("Should not be an error: %v", err)
+	}
+	if arn != "registerArn" {
+		t.Errorf("Wrong arn: %v", arn)
+	}
+}
+
+func TestValidateRegisteredAttributes(t *testing.T) {
```

Comment:
Can you add some attributes with no value set? As there are some attributes only contain name without value like: `"name": "com.amazonaws.ecs.capability.logging-driver.json-file"`I think the value returned of these attributes will be ""?